Lib/test/test_httplib.py

```diff
@@ -274,6 +274,45 @@ def test_ipv6host_header(self):
         conn.request('GET', '/foo')
         self.assertTrue(sock.data.startswith(expected))
 
+    def test_request_path_handling(self):
+        conn = client.HTTPConnection('server.fqdn')
```

Comment:
I believe there is too much duplication in the unit tests for both of your pull requests. Please use a loop or two (like what is done in `test_invalid_headers()`).

Reply:
Sure; done. Any preference on which approach to take, though?